lib/OpenQA/Worker/Engines/isotovideo.pm

```diff
@@ -326,11 +340,28 @@ sub engine_workit {
         my $error = locate_local_assets(\%vars, $assetkeys);
         return $error if $error;
     }
+    my $casedir             = OpenQA::Utils::testcasedir($vars{DISTRI}, $vars{VERSION}, $shared_cache);
+    my $productdir          = OpenQA::Utils::productdir($vars{DISTRI}, $vars{VERSION}, $shared_cache);
+    my $relative_casedir    = substr($casedir,          length($vars{PRJDIR}) + 1);
+    my $relative_productdir = substr($productdir,       length($vars{PRJDIR}) + 1);
+    my $target_name         = substr($relative_casedir, 0, index($relative_casedir, '/'));
```

Comment:
On the first look it seems a little bit odd to construct these absolute paths just to make them relative again using `substr` and `length`. Maybe it makes more sense to adjust the 2 utility functions so they can directly return paths relative to the share dir? Or add relative versions of them, e.g. `testcasedir_rel` and `productdir_rel`. The absolute versions could maybe be implemented in terms of the relative ones by simply adding the absolute prefix afterwards.

Note that this code is very sub-optimal anyways because `testcasedir` might return a path which is under `$prjdir/tests` and not under `$prjdir/share`. Your code unconditionally removes the first `length($vars{PRJDIR})` characters which works only in both cases because `tests` and `share` have coincidentally the same number of characters. It also means that the directory the relative paths are relative to depends on the outcome of `testcasedir`. However, I suppose it should always be relative to `sharedir`/`$vars{PRJDIR}`.

Maybe we could also drop the case where we look for tests under `$prjdir/tests` as is complicates things. @okurz What do you think?

Reply:
After committing the code, I did a test that enabled cache service. Using a relative path compared to `$prjdir` or `$share` is not correct in this scenario. For example:
The values in `vars.json` are
```
PRJDIR=/var/lib/openqa/share
CACHEDIRECTORY=/var/lib/openqa/cache
``` 

the $casedir (abosulte path ) is `/var/lib/openqa/cache/localhost/tests/sle`. so we cannot use PRJDIR or CACHEDIRECTORY to get the relative path.
How about keeping the last version modification, just symlink `$casedir` to `path($casedir)->basename`, so we do not need to care whether we enable or disable the cache service, and the `sharedir`.